### PR TITLE
prometheus example

### PR DESCRIPTION
**Pull request by esigo**
Adds example for  #1315 (issue)
depends on #1330 and #1331 

## Changes
Adds prometheus example.
Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/prometheus/BUILD [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    ],
+    linkopts = ["-pthread"],
+    tags = ["ostream"],
+    deps = [
+        "//api",
+        "//examples/common/metrics_foo_library:common_metrics_foo_library",
+        "//exporters/prometheus:prometheus_exporter",
+        "//sdk/src/metrics",
+    ],
+)
```
Comment (lalitb) [inline examples/prometheus/BUILD:14]: Can we add cmake build too, or else github issue to track if plan to add it separately ?

File: examples/prometheus/main.cc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#  include "opentelemetry/sdk/metrics/meter.h"
+#  include "opentelemetry/sdk/metrics/meter_provider.h"
+
+#  ifdef BAZEL_BUILD
+#    include "examples/common/metrics_foo_library/foo_library.h"
+#  else
+#    include "metrics_foo_library/foo_library.h"
+#  endif
+
+namespace metric_sdk      = opentelemetry::sdk::metrics;
```
Comment (lalitb) [inline examples/prometheus/main.cc:21]: nit - metrics_sdk ?

Comment (lalitb) [inline examples/prometheus/main.cc:21]: nit - metrics_sdk ?

File: examples/prometheus/main.cc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+namespace nostd           = opentelemetry::nostd;
+namespace common          = opentelemetry::common;
+namespace exportermetrics = opentelemetry::exporter::metrics;
```
Comment (lalitb) [inline examples/prometheus/main.cc:24]: nit - metrics_exporter ?

Comment (lalitb) [inline examples/prometheus/main.cc:24]: nit - metrics_exporter ?

File: examples/prometheus/main.cc [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+namespace
+{
+
+void initMetrics(const std::string &name)
+{
+  std::string addr{"localhost:8080"};
+  std::puts("PrometheusExporter example program running ...");
+
+  std::unique_ptr<metric_sdk::MetricExporter> exporter{
+      new exportermetrics::PrometheusExporter(addr)};
```
Comment (lalitb) [inline examples/prometheus/main.cc:36]: nit - need to update to use options.

Comment (lalitb) [inline examples/prometheus/main.cc:36]: nit - need to update to use options.

File: examples/prometheus/main.cc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  std::string schema{"https://opentelemetry.io/schemas/1.2.0"};
+
+  // Initialize and set the global MeterProvider
+  metric_sdk::PeriodicExportingMetricReaderOptions options;
+  options.export_interval_millis = std::chrono::milliseconds(1000);
+  options.export_timeout_millis  = std::chrono::milliseconds(500);
+  std::unique_ptr<metric_sdk::MetricReader> reader{
+      new metric_sdk::PeriodicExportingMetricReader(std::move(exporter), options)};
+  auto provider = std::shared_ptr<metrics_api::MeterProvider>(
+      new metric_sdk::MeterProvider(std::move(exporters)));
```
Comment (lalitb) [inline examples/prometheus/main.cc:49]: Not related to this PR but need to modify MeterProvider / MeterContext API to not use exporters directly, as they are configured through Reader. Will create a github issue for that.

Comment (lalitb) [inline examples/prometheus/main.cc:49]: Not related to this PR but need to modify MeterProvider / MeterContext API to not use exporters directly, as they are configured through Reader.

**Review by lalitb — APPROVED** [review]
LGTM once minor comments are fixed.

File: examples/prometheus/main.cc [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+namespace common          = opentelemetry::common;
+namespace exportermetrics = opentelemetry::exporter::metrics;
+namespace metrics_api     = opentelemetry::metrics;
+
+namespace
+{
+
+void initMetrics(const std::string &name)
+{
+  std::string addr{"localhost:8080"};
```
Comment (ThomsonTan) [inline examples/prometheus/main.cc:32]: Could this be configurable?

File: examples/prometheus/main.cc [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  p->AddView(std::move(histogram_instrument_selector), std::move(histogram_meter_selector),
+             std::move(histogram_view));
+  metrics_api::Provider::SetMeterProvider(provider);
+}
+}  // namespace
+int main(int argc, char **argv)
+{
+  if (argc != 2)
+  {
+    std::puts("invalid number of command line arguments");
```
Comment (ThomsonTan) [inline examples/prometheus/main.cc:78]: Output a line of example usage?

**Review by ThomsonTan — APPROVED** [review]

File: examples/prometheus/main.cc (hunk 2 above)
Comment (esigo) [reply examples/prometheus/main.cc]: thanks, fixed.

Comment (esigo) [reply examples/prometheus/main.cc]: thanks, fixed.

File: examples/prometheus/main.cc (hunk 3 above)
Comment (esigo) [reply examples/prometheus/main.cc]: thanks, fixed.

Comment (esigo) [reply examples/prometheus/main.cc]: thanks, fixed.

File: examples/prometheus/main.cc (hunk 4 above)
Comment (esigo) [reply examples/prometheus/main.cc]: thanks, fixed.

File: examples/prometheus/main.cc (hunk 7 above)
Comment (esigo) [reply examples/prometheus/main.cc]: thanks, added.

**Pull request merged**